README.md

```diff
@@ -31,8 +31,99 @@ transitive dependencies:
 `models`
 `testing` -> `api` -> `models`
 
-#### Adding Tests
-Tests are located in the `Tests/` directory. To add a new test, either write a new function in one of the existing classes, or create a new class and add the function into the `Tests/` directory.
+#### Writing Tests
+
+Tests are located in the `Tests` directory
+
+End-to-End tests often utilize the `MockServer` api to either assert against certain messages reaching the server or simply blocking until a request is sent/received
+
+A new `MockServer` instance is automatically initialized before every single test, as long as your test class extends one of the base classes
+
+`BaseTest`: a fresh `MockServer` instance, mParticle SDK has not already been started when the tests executes
+`BaseStartedTest`: a fresh `MockServer` instance, and a fully initialized (initial `identify()` call has completed`) mParticle SDK instance
+
+The `MockServer` instance is stored in a singleton, `server`. Tests should only interact with their `MockServer` instance through this singleton, attempting to invoke the class directly **will** lead to threading/freezing erros when run on iOS.
+
+##### Assertions
+Tests may use the `MockServer` to assert against either
+- "current" requests state (have/have not already received a matching request), or
+- "future" requests state (will/will not be received).
+
+For the latter case, you can also choose to block until the assertion passes or continue the test logic and check it's state at a later point. In order to avoid timing issues for "future" blocking assertions for "future" assertions, the test may also specify the "trigger" they wish to test in the `after { }` block
```

Comment:
```suggestion
For the future reuqests state, you can also choose to block until the assertion passes or continue the test logic and check its state at a later point. In order to avoid timing issues for "future" blocking assertions for "future" assertions, the test may also specify the "trigger" they wish to test in the `after { }` block
```

Reply:
I was confused by "for future blocked assertions for future assertions," but if a developer wouldn't be confused, no need to modify. But check and make sure that one of those 'assertions' isn't meant to be 'request'?

Reply:
Ah, I think was rephrasing that and ended up leaving duplicates in there.. I'm going to update it to just "for future assertions"